indico/modules/events/editing/controllers/base.py

```diff
@@ -7,31 +7,63 @@
 
 from __future__ import unicode_literals
 
-from flask import request
+from flask import request, session
+from werkzeug.exceptions import Unauthorized
 
 from indico.modules.events.contributions.controllers.display import RHContributionDisplayBase
 from indico.modules.events.controllers.base import RHDisplayEventBase
 from indico.modules.events.editing.models.editable import Editable, EditableType
+from indico.modules.events.editing.settings import editing_settings
 from indico.modules.events.management.controllers.base import RHManageEventBase
+from indico.modules.events.models.events import Event
 from indico.web.rh import RequireUserMixin
 
 
-class RHEditingBase(RequireUserMixin, RHDisplayEventBase):
+class TokenAccessMixin(object):
+    SERVICE_ALLOWED = False
+
+    def _token_can_access(self):
+        # we need to "fish"  the event here because at this point _check_params
+        # hasn't run yet
+        event = Event.get(int(request.view_args['confId']))
```

Comment:
`, is_deleted=False`, and I think right now we may also get a rather ugly error when accessing an URL using this mixin with an invalid event id that doesn't exist, `.get_one(...)` should avoid this.